engine/execution/storehouse/block_end_snapshot_test.go

```diff
@@ -0,0 +1,103 @@
+package storehouse_test
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+
+	executionMock "github.com/onflow/flow-go/engine/execution/mock"
+	"github.com/onflow/flow-go/engine/execution/storehouse"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func TestBlockEndSnapshot(t *testing.T) {
+	t.Run("Get register", func(t *testing.T) {
+		header := unittest.BlockHeaderFixture()
+
+		// create mock for storage
+		store := &executionMock.RegisterStore{}
```

Comment:
using `New*` automatically asserts all expectations
```suggestion
		store := executionMock.NewRegisterStore(t)
```